src/ucp/rma/basic_rma.c

```diff
@@ -281,6 +297,8 @@ ucs_status_t ucp_get_nbi(ucp_ep_h ep, void *buffer, size_t length,
     uct_rkey_t uct_rkey;
     ucs_status_t status;
 
+    UCP_THREAD_CS_ENTER(ep->worker->mt_config);
```

Comment:
It is already on line 300 now... could you specify which APIs you think lock should not be taken?